### ISSUE TITLE
Run an OHIF viewer Docker container built from the v3-stable Dockerfile does not work

### ISSUE THREAD

**Issue by jbocce** [OHIF/Viewers#3079]
### Describe the Bug

Attempting to run an OHIF viewer Docker container built using the Dockerfile contained in the v3-stable branch does not work. The building of the Docker image appears to work fine, but running the image to create the container does not work. The container does not run at all and the following  logs are output...

```
2022-12-14 16:42:30 /usr/src/entrypoint.sh: line 0: 3000: not found
2022-12-14 16:42:30 2022/12/14 21:42:30 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/conf.d/default.conf:11
2022-12-14 16:42:30 nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/conf.d/default.conf:11
2022-12-14 16:42:30 Starting Nginx to serve the OHIF Viewer...
```

### What steps can we follow to reproduce the bug?

1. Install Docker (desktop) from [here](https://docs.docker.com/get-docker/)
2. Clone the OHIF/Viewers repo and checkout the `v3-stable` branch.
3. At a terminal window at the root of the source tree, build a local OHIF Viewer Docker image using the command `docker build . -t ohif-viewer-bug:latest`. Note that what follows `-t` (i.e. tag) option is `{name}:{tag}` and is arbitrary for the case of this bug.
4. The Docker image builds fine. Now attempt to run a container from the image by issuing the command `docker run -d -p 3000:3000/tcp --name ohif-viewer-bug-container ohif-viewer-bug`
5. The container gets created but does NOT run. visiting http://localhost:3000 brings up a "Site cannot be reached" page. Looking at the Docker Desktop logs shows the error listed above.

### Other things to note

1. The v3-stable Dockerfile looks to run the `nginx` server inside the container on port 3000 by default. This is a little strange because the the default port is typically 80 and references to the [OHIF Viewer Docker image on Docker Hub](https://hub.docker.com/r/ohif/viewer) are using port 80. Consideration should be given to changing it back to 80.
2. The v3-stable Dockerfile looks to making the `nginx` server port, inside the container, configurable. It does so by substituting the value of the ${PORT} environment variable in the `nginx` configuration file using the `envsubst` command. However, this looks to be done twice: once in the Dockerfile and again in the `entrypoint.sh` script file. It makes most sense to just have it in one place and even more sense to just have it in the Docker entry point script file so that it can be changed per container instance..

**Comment by mr-sonalia** [comment]
I am facing an issue related to the docker image itself. 
I have an Orthanc server set up as a docker container. When I run ohif locally, it works as expected.
But, when I build the docker image, it does not fetch the data from my orthanc server, instead it falls back to its default orthanc server.

**Comment by jbocce** [comment]
@mr-sonalia thanks for the question. One way is to set the APP_CONFIG environment variable when the docker image is built. So in the Dockerfile try something like the following...

```
RUN APP_CONFIG=config/local_orthanc.js \
yarn run build
```
Hope this helps.

**Comment by mr-sonalia** [comment]
@jbocce  Thanks for the response, what would be the content of local_orthanc.js file?

**Comment by jbocce** [comment]
Hi @mr-sonalia.

`local_othanc.js` can be found in the v3-stable branch [here](https://github.com/OHIF/Viewers/blob/v3-stable/platform/viewer/public/config/local_orthanc.js). I suspect your OHIF configuration might look similar to this?

But let me ask a question for clarification... Are you trying to run an OHIF Docker container that fetches data from an Orthanc Docker container?

**Comment by mr-sonalia** [comment]
@jbocce , yes my OHIF configuration looks similar to the one you've shared.

Yes exactly, I want my OHIF container to fetch data from my Orthanc container.

**Comment by jbocce** [comment]
@mr-sonalia - All I believe you have to do is change the Dockerfile as I mentioned before. If the path (including the file name too) to your configuration file is {the path}, then the Dockerfile should have something like the following...

```
RUN APP_CONFIG= {the path} \
yarn run build
```

**Comment by mr-sonalia** [comment]
@jbocce , thanks, I will try this asap!

**Comment by mr-sonalia** [comment]
@jbocce you have suggested adding `RUN APP_CONFIG={the path} \`, wouldn't it be `ENV ...`?

**Comment by jbocce** [comment]
> @jbocce you have suggested adding `RUN APP_CONFIG={the path} \`, wouldn't it be `ENV ...`?

The setup I have does not use `ENV`, but feel free to experiment with it. What I suggested sets the `APP_CONFIG` environment variable so that it is available to the OHIF build.

Perhaps I confused things a bit, but the config file should be in the same directory as the `local_orthanc.js` one I pointed out earlier. So if you config file is say named `mr_sonalia.js` then it should be in the config directory and the build line(s) in the Dockerfile should be something like this...

```
RUN APP_CONFIG=config/mr_sonlia.js \
yarn run build
```

My setup works with the above. Good luck with it.

**Comment by mr-sonalia** [comment]
@jbocce thanks, just to be sure, the only change that you made was in the root Dockerfile, right? Because I got this response from the server: 
![image](https://user-images.githubusercontent.com/53109900/215672694-da61d372-2d0b-461e-a854-541c52d33ae2.png)
![image](https://user-images.githubusercontent.com/53109900/215672636-defe9941-52e7-408c-a952-a9819fb2cdc3.png)

Also, this is my docker-compose configuration, 
```
  ohif:
    image: ohif-v3
    depends_on:
      - orthanc
    container_name: ohif-v3
    ports:
      - "3000:8080/tcp"
    environment:
      - PORT=8080
      - NODE_ENV=production
```

So sorry to bother you again!

**Comment by jbocce** [comment]
@mr-sonalia. Yes the only changes I made were in the Dockerfile. I was running two containers: orthanc and ohif. It looks to me that you were composing the two. So I just tried doing  a compose with the SAME edits to the Dockerfile and it worked for me as well. 

Is the `ohif-v3` image listed in the docker-compose that you sent built as per these [instructions](https://v3-docs.ohif.org/deployment/docker/)? I am sorry but I really don't know how else to help you.
